### ISSUE TITLE
Configurable products not deducting stock properly

### ISSUE THREAD

**Issue by BrentRobert** [AmpersandHQ/magento2-disable-stock-reservation#57]
The change underneath resulted in the creation of multiple itemRequests for configurableProducts.
https://github.com/AmpersandHQ/magento2-disable-stock-reservation/commit/5e46cef6788996b7bc30a4bfaa805de1481fcb38

When ordering a configurable product, a row is created for the simple and the configurable. However Magento uses the SKU of the simple for both on the order item. The mapping that was created above results in allowing configurable order items to be deducted on the SKU of the simple.

The reason this has been unnoticed is because the itemRequests are not grouped as in 'GetItemsToDeductFromOrder'. Magento loops over these requests and overwrites SKU based in the array. Always using the latest value. In this case the simple.

```
foreach ($inventoryRequest->getItems() as $item) {
    $normalizedSku = $this->normalizeSku($item->getSku());
    $itemsTdDeliver[$normalizedSku] = $item->getQty();
}
```

TLDR;
- The fix for grouped products actually broke the configurable ones.
- The itemRequests should be grouped by SKU

**Comment by tr33m4n** [comment]
Hi @BrentRobert, is this issue now resolved? You had a PR into the module but it's since been closed. Thanks

**Comment by BrentRobert** [comment]
@tr33m4n I honestly have no idea. I closed the PR as it had been open for 2 years.
